### ISSUE TITLE
Cannot install. `python setup.py install` fails

### ISSUE THREAD

**Issue by leotrs** [jonschwenk/RivGraph#38]
Hello. I am following the instructions for [building from source](https://jonschwenk.github.io/RivGraph/install/index.html#installation-from-source). 

I believe these instructions should mention that the following command is necessary before step 2: `pip install -r requirements.txt`. If so, then the `requirements.txt` file should be updated as it currently does not work:
```
$ pip install -r requirements.txt 
Collecting numpy
  Using cached numpy-1.19.5-cp38-cp38-manylinux2010_x86_64.whl (14.9 MB)
Collecting scikit-image
  Using cached scikit_image-0.18.1-cp38-cp38-manylinux1_x86_64.whl (30.2 MB)
ERROR: Could not find a version that satisfies the requirement opencv (from -r requirements.txt (line 3)) (from versions: none)
ERROR: No matching distribution found for opencv (from -r requirements.txt (line 3))
(rivgraph) leo@syenite: RivGraph [master]$ pip install opencv
ERROR: Could not find a version that satisfies the requirement opencv (from versions: none)
ERROR: No matching distribution found for opencv
```
Currently, requirements.txt lists `opencv`. I believe the correct dependency is `opencv-python`.  After changing `opencv` for `opencv-python`, the command `pip install -r requirements.txt` runs smoothly. (Though it should perhaps be made to match `environment.yml`.)

After that, I tried the following:
```
$ python setup.py install
Traceback (most recent call last):
  File "setup.py", line 2, in <module>
    from rivgraph import __version__
  File "/home/leo/code/RivGraph/rivgraph/__init__.py", line 3, in <module>
    from rivgraph.classes import delta, river
  File "/home/leo/code/RivGraph/rivgraph/classes.py", line 9, in <module>
    import gdal
ModuleNotFoundError: No module named 'gdal'
```

I tried running `apt install libgdal-dev`, which executed successfully. However, the command `pip install gdal` now fails.

As a result, the command `python setup.py install` fails with error `ModuleNotFoundError: No module named 'gdal'`.

-----------

As a sidenote, I tried the installation with conda, and it also failed. This may have nothing to do with your package as I have never used conda before. Note I would **much** rather be able to install from source than having to download and use conda. Requiring conda adds 500MB+ of space required.

For completeness, after downloading, installing, and verifying my installation of conda, I got the following error:

```
$ conda env create --file environment.yml

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda/exceptions.py", line 1079, in __call__
        return func(*args, **kwargs)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/cli/main.py", line 80, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/cli/main_create.py", line 87, in execute
        spec = specs.detect(name=name, filename=filename, directory=os.getcwd())
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/specs/__init__.py", line 43, in detect
        if spec.can_handle():
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/specs/yaml_file.py", line 18, in can_handle
        self._environment = env.from_file(self.filename)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/env.py", line 160, in from_file
        return from_yaml(yamlstr, filename=filename)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda_env/env.py", line 141, in from_yaml
        data = yaml_safe_load(yamlstr)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/conda/common/serialize.py", line 67, in yaml_safe_load
        return yaml.safe_load(string, version="1.2")
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/main.py", line 980, in safe_load
        return load(stream, SafeLoader, version)
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/main.py", line 935, in load
        return loader._constructor.get_single_data()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/constructor.py", line 109, in get_single_data
        node = self.composer.get_single_node()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/composer.py", line 78, in get_single_node
        document = self.compose_document()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/composer.py", line 104, in compose_document
        self.parser.get_event()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/parser.py", line 163, in get_event
        self.current_event = self.state()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/parser.py", line 239, in parse_document_end
        token = self.scanner.peek_token()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/scanner.py", line 182, in peek_token
        self.fetch_more_tokens()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/scanner.py", line 282, in fetch_more_tokens
        return self.fetch_value()
      File "/home/leo/anaconda3/lib/python3.8/site-packages/ruamel_yaml/scanner.py", line 651, in fetch_value
        raise ScannerError(
    ruamel_yaml.scanner.ScannerError: mapping values are not allowed here
      in "<unicode string>", line 90, column 65:
         ... e" content="{&quot;version&quot;: &quot;4&quot;, &quot;rollouts& ... 
                                             ^ (line: 90)

`$ /home/leo/anaconda3/bin/conda-env create --file environment.yml`

  environment variables:
                 CIO_TEST=<not set>
  CONDA_AUTO_UPDATE_CONDA=false
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/home/leo/anaconda3/bin/conda
             CONDA_PREFIX=/home/leo/anaconda3
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/home/leo/anaconda3/bin/python
               CONDA_ROOT=/home/leo/anaconda3
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
                     PATH=/home/leo/anaconda3/bin:/home/leo/anaconda3/bin:/home/leo/anaconda3/co
                          ndabin:/home/leo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
                          :/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
               WINDOWPATH=2

     active environment : base
    active env location : /home/leo/anaconda3
            shell level : 1
       user config file : /home/leo/.condarc
 populated config files : /home/leo/.condarc
          conda version : 4.9.2
    conda-build version : 3.20.5
         python version : 3.8.5.final.0
       virtual packages : __glibc=2.31=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/leo/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/leo/anaconda3/pkgs
                          /home/leo/.conda/pkgs
       envs directories : /home/leo/anaconda3/envs
                          /home/leo/.conda/envs
               platform : linux-64
             user-agent : conda/4.9.2 requests/2.24.0 CPython/3.8.5 Linux/5.6.0-1042-oem ubuntu/20.04.1 glibc/2.31
                UID:GID : 1001:1001
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```

**Comment by jonschwenk** [comment]
Thanks for the report, @leotrs.

The requirements.txt file was outdated--apologies. We have removed it entirely, and now only point now to the enviornment.yml file.

@elbeejay has updated the documentation (see PR #39 ) for installation from source to include links to each of the dependency repos--this should take care of the mismatched names (e.g. opencv). 

As noted in that document, gdal is notoriously difficult sometimes to install properly. We can't really provide support for that, as the problems are too difficult to reproduce on different systems (and can be due to a number of things--paths seem to be one of the main culprits).

As for the conda installation, we use this method frequently and have not encountered the issue you report. The error message suggests something may be amiss in the .yml file, but again, we use this same file for our Github Actions builds without issue. 

Perhaps one thing to try that might "kill two birds" would be to install miniconda (instead of Anaconda). It takes up far less space, and perhaps a fresh install might solve the conda install issue.

@elbeejay please chime in with any additional notes (or correct me where I'm wrong).

**Comment by leotrs** [comment]
Ok, I was finally able to install. The above error is totally my fault! Apologies.

Here's what happened in [the instructions](https://jonschwenk.github.io/RivGraph/install/index.html), it says to `(right-click, "Save As...")`. I understood that too literally and actually right-clicked _the link_ and chose "Save link as...". However, this saves the HTML source code of the page the link points to, so I actually saved the HTMl of [this page](https://github.com/jonschwenk/RivGraph/blob/master/environment.yml). I didn't realize this and just executed `$ conda env create --file /path/to/environment.yml`, so I was giving it the wrong file.

What I had to do was go to [this page](https://github.com/jonschwenk/RivGraph/blob/master/environment.yml), click on "Raw", and copy&save the contents manually to an `environment.yml` file. And now `$ conda env create --file /path/to/environment.yml` works perfectly. This is with a fresh installation of miniconda, though I'm sure it would also have worked with my previous attempt using conda.

BTW, this is on Chrome/Ubuntu.
